tests/metrics/test_categorical_risk_observer.py

```diff
@@ -0,0 +1,141 @@
+from vivarium_public_health.metrics.stratification import (
+    ResultsStratifier as ResultsStratifier_,
+)
+from vivarium_public_health.risks.base_risk import Risk
+from vivarium_public_health.utilities import to_years
+
+
+@pytest.fixture
+def categorical_risk():
+    year_start = 1990
+    year_end = 2010
+    risk = "test_risk"
+    risk_data = dict()
+    exposure_data = build_table(
+        0.25, year_start, year_end, ["age", "year", "sex", "cat1", "cat2", "cat3", "cat4"]
+    ).melt(
+        id_vars=("age_start", "age_end", "year_start", "year_end", "sex"),
+        var_name="parameter",
+        value_name="value",
+    )
+
+    risk_data["exposure"] = exposure_data
+    risk_data["categories"] = {
+        "cat1": "severe",
+        "cat2": "moderate",
+        "cat3": "mild",
+        "cat4": "unexposed",
+    }
+    risk_data["distribution"] = "ordered_polytomous"
+    return Risk(f"risk_factor.{risk}"), risk_data
+
+
+# Subclass of ResultsStratifier for integration testing
+class ResultsStratifier(ResultsStratifier_):
+    configuration_defaults = {
+        "stratification": {
+            "default": ["age_group", "sex"],
+        }
+    }
+
+
+def test_observation_registration(base_config, base_plugins, categorical_risk):
+    """Test that all expected observation keys appear as expected in the results."""
+    risk, risk_data = categorical_risk
+    observer = CategoricalRiskObserver(f"{risk.risk.name}")
+    simulation = InteractiveContext(
+        components=[
+            TestPopulation(),
+            ResultsStratifier(),
+            risk,
+            observer,
+        ],
+        configuration=base_config,
+        plugin_configuration=base_plugins,
+        setup=False,
+    )
+    simulation.configuration.update(
+        {
+            "stratification": {
+                "test_risk": {
+                    "exclude": ["age_group"],
+                }
+            }
+        }
+    )
+
+    for key, value in risk_data.items():
+        simulation._data.write(f"risk_factor.test_risk.{key}", value)
+
+    simulation.setup()
+    pop = simulation.get_population()
+    simulation.step()
+    results = simulation.get_value("metrics")
+    expected_observations = [
+        "MEASURE_test_risk_cat1_person_time_SEX_Female",
+        "MEASURE_test_risk_cat1_person_time_SEX_Male",
+        "MEASURE_test_risk_cat2_person_time_SEX_Female",
+        "MEASURE_test_risk_cat2_person_time_SEX_Male",
+        "MEASURE_test_risk_cat3_person_time_SEX_Female",
+        "MEASURE_test_risk_cat3_person_time_SEX_Male",
+        "MEASURE_test_risk_cat4_person_time_SEX_Female",
+        "MEASURE_test_risk_cat4_person_time_SEX_Male",
+    ]
+
+    assert set(expected_observations) == set(results(pop.index).keys())
+
+
+def test_observation_correctness(base_config, base_plugins, categorical_risk):
+    """Test that person time appear as expected in the results."""
+    time_step = pd.Timedelta(days=base_config.time.step_size)
```

Comment:
Is there anything different between the simulation in this test and the one above it? If not, we should either combine them into the same test or create a fixture so we only have to create and run these simulations once (to reduce the runtime of our test suite).